### PR TITLE
feat(query): tools for decoding the new table meta data 

### LINKED ISSUE #11200: feature request: tools for decoding the new table meta data
> BTW, do we have a tool or functions to decode the compressed bincode meta file?Like we want to check the meta data in it, for json it's easy to do it.

Command-line tool name: databend-meta-decoder

Usage: 
`databend-meta-decoder  -i <INPUT> [-j]  [-o <OUTPUT>] -t <TYPE>`

Description:
Decode and output the encoded metadata to the terminal or file. Supports bincode, MessagePack, and JSON encoding formats.

Options:
-j, --json Output data in JSON format
-i, --input <INPUT> Specify the file path to be decoded
-o, --output <OUTPUT> Specify the output file path for the decoded data. If not specified, output will be displayed on the terminal
-t, --type <TYPE> input data type: segment or snapshot(type only support `sg` or `ss`)

Examples:

1. View metadata information in binary format without saving to a file:
```shell
databend-meta-decoder -i data.bin

Version: x
Encoding: y
Compression: z
Blocks Size: p bytes
Summary Size: q bytes
```

2. Convert metadata information in binary format to JSON format and save it to a file:
```shell
databend-meta-decoder -i data.bin -j -o output.json
```

Note:
If the output file already exists, it will be overwritten.

How about this design?

_Originally posted by @jun0315 in https://github.com/datafuselabs/databend/issues/11015#issuecomment-1505628486_

### PR DESCRIPTION
**Pull request by jun0315**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Command-line tool name: table-meta-inspector

Usage:
`
table-meta-inspector  [-i <INPUT>]  [-o <OUTPUT>] -t <TYPE> [-c <CONFIG>]
`
Description:
Decode and output the encoded metadata to the terminal or file. Supports bincode, MessagePack, and JSON encoding formats.

Options:
-i, --input Specify the file path to be decoded, if not specified, it will use os.stdin.
-o, --output Specify the output file path for the decoded data. If not specified, it will use os.stdout.
-t, --type input data type: segment or snapshot(type only support sg or ss)
-c, --config config file path,query-config.toml, can read storage of config.

Examples:
1. 
```bash
[qijun@databend-arch databend]$ ./target/debug/table-meta-inspector -c scripts/ci/deploy/config/databend-query-node-1.toml -i 1/11/_sg/613f7483c61243c1a57419211978ee38_v3.bincode -t sg
```
```bash
[qijun@databend-arch databend] ${"format_version":3,"blocks":[{"row_count":1,"block_size":4,"file_size":192,"col_stats":{"0":{"min":{"Number":{"Int32":1}},"max":{"Number":{"Int32":1}},"null_count":0,"in_memory_size":4,"distinct_of_values":1}},"col_metas":{"0":{"Parquet":{"offset":4,"len":35,"num_values":1}}},"cluster_stats":null,"location":["1/11/_b/15374f874606444f9a4e961cf3e94ec5_v2.parquet",2],"bloom_filter_index_location":["1/11/_i_b_v2/15374f874606444f9a4e961cf3e94ec5_v4.parquet",4],"bloom_filter_index_size":276,"compression":"Zstd"}],"summary":{"row_count":1,"block_count":1,"perfect_block_count":0,"uncompressed_byte_size":4,"compressed_byte_size":192,"index_size":276,"col_stats":{"0":{"min":{"Number":{"Int32":1}},"max":{"Number":{"Int32":1}},"null_count":0,"in_memory_size":4,"distinct_of_values":null}}}}
```
2. 
```bash
[qijun@databend-arch databend]$ cat .databend/stateless_test_data/1/11/_sg/613f7483c61243c1a57419211978ee38_v3.bincode | ./target/debug/table-meta-inspector -t sg
```

```bash
[qijun@databend-arch databend] ${"format_version":3,"blocks":[{"row_count":1,"block_size":4,"file_size":192,"col_stats":{"0":{"min":{"Number":{"Int32":1}},"max":{"Number":{"Int32":1}},"null_count":0,"in_memory_size":4,"distinct_of_values":1}},"col_metas":{"0":{"Parquet":{"offset":4,"len":35,"num_values":1}}},"cluster_stats":null,"location":["1/11/_b/15374f874606444f9a4e961cf3e94ec5_v2.parquet",2],"bloom_filter_index_location":["1/11/_i_b_v2/15374f874606444f9a4e961cf3e94ec5_v4.parquet",4],"bloom_filter_index_size":276,"compression":"Zstd"}],"summary":{"row_count":1,"block_count":1,"perfect_block_count":0,"uncompressed_byte_size":4,"compressed_byte_size":192,"index_size":276,"col_stats":{"0":{"min":{"Number":{"Int32":1}},"max":{"Number":{"Int32":1}},"null_count":0,"in_memory_size":4,"distinct_of_values":null}}}}
```

Note:
If the output file already exists, it will be overwritten.

Closes #11200

### CODE REVIEW

File: src/binaries/table-meta/main.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Datafuse Labs.
```
Comment (BohuTANG) [inline src/binaries/table-meta/main.rs:1]: Better to name `src/binaries/tools/table_meta_inspector.rs`, then we can add others tool to here in future.

Comment (jun0315) [reply src/binaries/table-meta/main.rs:1]: get it!

File: src/binaries/tool/table_meta_inspector.rs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    file.write_all(&out)?;
+
+    println!("Output file: {:?}", output_path);
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let config = InspectorConfig::parse();
+    // let args = App::new("table-meta-inspector")
```
Comment (sundy-li) [inline src/binaries/tool/table_meta_inspector.rs:122]: useless codes should be erased.

File: src/binaries/tool/table_meta_inspector.rs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+use serde::Serialize;
+use serde_json;
+use serfig::collectors::from_file;
+use serfig::parsers::Toml;
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Parser)]
+#[clap(about, version = &**DATABEND_COMMIT_VERSION, author)]
+pub struct InspectorConfig {
+    #[clap(long, short = 'i')]
+    pub input: String,
```
Comment (sundy-li) [inline src/binaries/tool/table_meta_inspector.rs:42]: input could be os.stdin if none.

So we could use shell pipe like


```
  cat xxx.bin   |   ./inspecter    |  jq   > a.json
```

**Review by sundy-li — APPROVED** [review]

**Pull request merged**